docs/advanced/pool.rst

```diff
@@ -451,22 +451,35 @@ Since `!psycopg_pool` version 3.3, the `!close_returns` parameter of the pool
 allows to change this behaviour, telling its connections to return to the
 pool, instead of closing, when `!close()` is called.
 
-For example:
+For example, the initialization in an async application may look like:
 
 .. code:: python
 
-    import sqlalchemy.pool
     import psycopg_pool
+    import sqlalchemy.pool
+    import sqlalchemy.ext.asyncio
 
     db_url = "postgresql://postgres:postgres@hostname:port/database"
 
-    pgpool = psycopg_pool.ConnectionPool(conninfo=db_url, close_returns=True)
+    async def main():
+        # Create and open a Psycopg pool
+        pgpool = psycopg_pool.AsyncConnectionPool(
+            conninfo=db_url, open=False, close_returns=True
+        )
+        await mypool.open()
 
-    engine = sqlalchemy.create_engine(
-        url=db_url.replace("postgresql://", "postgresql+psycopg://"),
-        poolclass=sqlalchemy.pool.NullPool,
-        creator=pgpool.getconn,
-    )
+        # Create a SQLAlchemy engine using the pool just created
+        engine = sqlalchemy.ext.asyncio.create_async_engine(
+            url=db_url.replace("postgresql://", "postgresql+psycopg://"),
```

Comment:
since you pass a creator you can also just hardcode `url="postgresql+psycopg://"`